R/grid.R

```diff
@@ -65,13 +65,32 @@ draw_custom_grid <- function(object, view.name){
   grid.args <- remove_field(grid.args, "equilogs")
   at <- list()
   for (side.name in side.names){
-    usr.at <- axis_axTicks(object, as.side(side.name))
-    if (is.null(usr.at))
-      at[[side.name]] <- grid_axTicks(object, as.side(side.name))
-    else
-      at[[side.name]] <- usr.at
+    side.axis <- as.axis(side.name)
+    if(side.axis == "x") {
+      n.side <- grid.args$nx 
+    } else { 
+      n.side <- grid.args$ny
+    }
+    
+    if(is.null(n.side)){
+      usr.at <- axis_axTicks(object, as.side(side.name))
+      if (is.null(usr.at)) {
+        at[[side.name]] <- grid_axTicks(object, as.side(side.name))
+      } else {
+        at[[side.name]] <- usr.at
+      }
+    } else {
+      plot.lims <- lim(object, as.side(side.name))
+      if(is.na(n.side)) {
+        at[[side.name]] <- NULL
+      } else {
+        at[[side.name]] <- seq.int(plot.lims[1], plot.lims[2], length.out = n.side + 1)
```

Comment:
I don't have time to do this today, but I think we want to reverse engineer how nx,ny works in base and follow that (ie calling axTicks or something like that). That being said, we could merge this pull request, and make an issue that nx/ny doesn't work on non-integers if that would help for repgen